### ISSUE TITLE
PDF thumbnail preview

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#15097]
**Is your feature request related to a problem? Please describe.**

I have a client who is giving me PDF magazines to upload to their site. 
First of all I am not sure of a clear way forward to show PDF links or thumbnails.
I tried the Media & Text block but that does not work with PDF files.

**Describe the solution you'd like**

I would like an easy way to add a pdf file and a preview of the front cover.
Perhaps using the Media & Text block. We have blocks for images, galleries, audio and we can also easily embed videos. As there is no general Media block to where we can add any media that also can create a preview. Perhaps we need a PDF block to where we can choose a link or preview thumbnail.

**Describe alternatives you've considered**

I end up following this process:
Open the PDF and take a screenshot of the front cover. Bring the image into Photoshop and readjust the size. Upload the PDF and front cover jpg. Add the jpg as an image. Then add a custom link to the PDF file.

Associated issues: 
https://github.com/WordPress/gutenberg/issues/8322
https://github.com/WordPress/gutenberg/issues/2195

**Comment by jorgefilipecosta** [comment]
Hi @paaljoachim, would use the columns block and adding each image in a column fit your use case?

**Comment by paaljoachim** [comment]
Hello Jorge

I just tested columns and that works fine. Thanks for the reminder!
Have a great evening!

**Comment by paaljoachim** [comment]
Bottom line is having a simple way to have a thumbnail preview of a PDF then using the preview to link directly to the PDF.

**Comment by idea--list** [comment]
Yes, placing PDF the expected way is a too complicated for the average user.
I guess designers thought of the "image" or "cover" block type for placing pdfs in the content.
However in that case you need to place a "custom link" on them and for that you need to know where to grab that link (most users tend not to be able to find that in the media library...).

As mentioned there is a built-in automatic cover image generation for PDF files but that works only if the server uses imagick. But even in that case it is not straightforward to show them up using GB. In that case why actually generating cover images automatically?

Another solution might be the "file" block type but that just places a text link and a DL button.

Or one could consider to use PDF-viewing services but those send cookies...so you need to update your cookie-policy in advance...

Would be really nice to solve all that with a toggle called "show cover" in the right sidebar for the file block type. That way there are no cookies involved and the average user does not get lost looking for the link while needing to click 5-6 times here and there.

**Comment by wpfangirl** [comment]
It seems like  a mistake not to make use of PDF thumbnails, since WordPress is already generating them. It really shouldn't require two steps to insert a PDF thumbnail that links directly to the PDF. In the Classic Editor you can select a thumbnail size when inserting a PDF (at least, you can if you have the Display All Image Sizes plugin installed). 

You can't use an image block for PDFs because the PDFs don't show up at all in the image block (reasonably, I suppose, given they are not images, but tweaking that block to allow PDFs in addition to images could be a solution).

That makes the File block pretty much the only option for inserting a PDF. I tried inserting the PDF thumbnail as an inline image in a file block. The result is horrible and it requires too many steps.

I ended up creating an ACF block to display PDFs with their thumbnails, and that will work fine for me, but somehow, with 15 gazillion different embed blocks (way more complexity than the classic editor), we don't have one for "document." 

Many of my clients produce hundreds of PDFs and want them on their websites. They shouldn't have to go through multiple steps to add a linked thumbnail of the document to a page or post. It wasn't optimal in the classic editor, but it's a veritable nightmare now, and an example of a place where the block editor provides **less** functionality than the classic editor.

**Comment by jsiren** [comment]
My use case is an online archive of past magazine issues, and similar material issued in PDF format. What I ended up doing manually is very simply put in an algorithm:

- Inside the initial **a** tag, insert an **img** tag.
- For the **img** tag, use the preview image URL for the **href** attribute, and add a class="pdfthumb" attribute for future formatting.
 - The thumbnail image URL can be formed simply: take the file URL, remove the final **.pdf**, and append **-pdf-{x}-{y}.jpg**, where {x} and {y} are the thumbnail dimensions.

This is a simplistic case where the image is simply a part of the existing link with no additional structure, and only a CSS class added for formatting. However, this is perfectly fine for my use case - I don't need a full-blown media management library, embedding of a hundred formats, etc.

**Comment by jsiren** [comment]
An alternative solution would be to amend the image block to support PDFs: the image on the page would be the thumbnail, and it would simply link to the PDF file.

**Comment by wpfangirl** [comment]
What happened to the PDF thumbnail image? It was in a version I tested, but didn't get added to the block that shipped in 5.8. PDF embeds are awkward and not accessible, and very rarely what I want. PDF thumbnails are already auto-generated by WordPress and available via [`wp_get_attachment_image()`](https://developer.wordpress.org/reference/functions/wp_get_attachment_image/).

**Comment by paaljoachim** [comment]
I am hoping to see inner blocks for the File block taken care of by @aurooba in this issue:
**File Block: file name needs to use PlainText component** 
https://github.com/WordPress/gutenberg/issues/10967

Having inner blocks will make it easier to extend the options. Hopefully also tackling replacing the file with a placeholder image.

**Comment by hartsook** [comment]
Perhaps a related issue? The UI for selecting the auto-generated pdf thumbnails needs help. In 5.8 and 5.9 I don't see the thumbnails in the Media LIbrary, and for example I can't choose a pdf thumbnail as the featured image for a Post that features that pdf. The 4 thumbnails are there in the /uploads folder but are not available through the expected image selection UIs.

**Comment by wpfangirl** [comment]
@hartsook It's never been possible to use the PDF image as the featured image. It doesn't respond to things like `get_the_post_thumbnail()` because you have to use `wp_get_attachment_image()`. To get around that, I've used the PDF Thumbnails plugin (premium), which generates separate images linked to the PDF file.

**Comment by bozzmedia** [comment]
I appreciate the efforts on this to date. Facing this issue in 2023, any hope of PDF thumbnails for the File block?

**Comment by idea--list** [comment]
Let me rectify something important: there is no evidence of any effort for 4 years now.
Actually it does not bother me as since that time i almost completely stopped using WP due to GB and the fact that it breaks every month in another way. Meanwhile disclosed vulnerabilities of plugins skyrockets in the past 6 months now. That was never the case before GB.

**Comment by jdaviescoates** [comment]
I makes me sad that 6 years on there is still no way to easily replicate things like this with a standard WordPress block:

<img width="973" height="495" alt="Image" src="https://github.com/user-attachments/assets/608b683f-b834-431b-b0ff-876f67f1aeb6" />

It's quite a common need for any organisation that regularly publishes/ shares documents. Does anyone know of any good solutions?

In the olden days I used to use this plugin https://wordpress.org/plugins/document-gallery/ but it's not been updated for years.
